ultralytics/nn/autobackend.py

```diff
@@ -199,7 +200,22 @@ def __init__(
                 device = torch.device("cpu")
                 cuda = False
             LOGGER.info(f"Preferring ONNX Runtime {providers[0]}")
-            session = onnxruntime.InferenceSession(w, providers=providers)
+            if imx:
```

Comment:
@ambitious-octopus `if else` statements should always have the most commonly expected first result first. In this case that's `if onnx`, NOT `if imx`